pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -273,14 +280,20 @@ func newVSphere(cfg VSphereConfig) (*VSphere, error) {
 		cfg.Global.RoundTripperCount = RoundTripperDefaultCount
 	}
 
-	c, err := newClient(context.TODO(), &cfg)
-	if err != nil {
-		return nil, err
-	}
+	var c *govmomi.Client
+	var id string
+	if cfg.Global.VMName == "" {
```

Comment:
Comment like this ```// Need credentials to obtain VMName``` could be more helpful. Also, I would to suggest to add log message here